extension/src/test/XLIFFTypes.test.ts

```diff
@@ -484,38 +485,66 @@ suite("Xliff Types - Functions", function () {
     );
   });
 
-  test("getSameSourceDifferentTarget", function () {
+  test("Xliff.getSameSourceDifferentTarget", function () {
     const xlf = Xliff.fromString(xliffXmlWithDuplicateSources());
     const transUnits = xlf.getSameSourceDifferentTarget(xlf.transunit[1]);
-    assert.equal(
+    assert.deepStrictEqual(
       transUnits.length,
       1,
       "Unexpected number of trans-units returned."
     );
   });
 
-  test("differentlyTranslatedTransunits", function () {
+  test("Xliff.differentlyTranslatedTransunits", function () {
     const xlf = Xliff.fromString(xliffXmlWithDuplicateSources());
     const transUnits = xlf.differentlyTranslatedTransUnits();
-    assert.notEqual(
+    assert.notDeepStrictEqual(
       transUnits.length,
       xlf.transunit.length,
       "Same number of transunit as the total was returned. No bueno!"
     );
-    assert.equal(
+    assert.deepStrictEqual(
       transUnits.length,
       3,
       "Unexpected number of transunits returned."
     );
     const id = transUnits.map((t) => {
       return t.id;
     });
-    assert.equal(
+    assert.deepStrictEqual(
       id.length,
       new Set(id).size,
       "Duplicate trans-units in result"
     );
   });
+
+  test("Xliff.getXliffIdTokenArray()", function () {
+    const langXlf = Xliff.fromString(getXliffMissingXliffGeneratorNote());
+    const unitMissingGeneratorNote = langXlf.getTransUnitById(
+      "Page 2931038265 - NamedType 12557645"
+    );
+    const normalUnit = langXlf.getTransUnitById(
+      "Table 2328808854 - NamedType 12557645"
+    );
+
+    assert.throws(
+      () => unitMissingGeneratorNote.getXliffIdTokenArray(),
+      /Could not find a note from "Xliff Generator" in trans-unit "Page 2931038265 - NamedType 12557645"/
+    );
+    assert.doesNotThrow(
+      () => normalUnit.getXliffIdTokenArray(),
+      /Could not find a note from "Xliff Generator" in trans-unit "Table 2328808854 - NamedType 12557645"/
```

Comment:
Why the heck does `doesNotThrow` take a message as a parameter?! 🙄 I just don't get it...

Reply:
My guess is that the function could throw another error? Someday I should look into how to properly work with errors in TS/JS :)